daft/expressions/expressions.py

```diff
@@ -428,6 +428,16 @@ def ceil(self) -> Expression:
         expr = self._expr.ceil()
         return Expression._from_pyexpr(expr)
 
+    def dot(self, other: Expression) -> Expression:
```

Comment:
thoughts on also overloading `@` like in numpy?

```
In [6]: np.ones(10) @ np.ones(10)
Out[6]: 10.0
```

Reply:
what about nested/list dtypes? do we just throw an unsupported error if you try to matmul on list types? 

numpy appears to error
```py
a = [[1, 2], [3, 4]]
b = [[5, 6], [7, 8]]

print(a @ b)
# TypeError: unsupported operand type(s) for @: 'list' and 'list'
```

Reply:
would we want to support nested dtypes too? 

```py
a = np.array([[1, 2], [3, 4]])
b = np.array([[5, 6], [7, 8]])

print(a @ b)
# [[19 22]
# [43 50]]